shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ResultSetUtil.java

```diff
@@ -77,6 +82,49 @@ public static Object convertValue(final Object value, final Class<?> convertType
             return value;
         }
     }
+    
+    private static Object convertURL(final Object value) {
+        String val = value.toString();
+        try {
+            return new URL(val);
+        } catch (MalformedURLException mfe) {
+            throw new ShardingSphereException("Unsupported Date type: URL");
+        }
+    }
+    
+    /**
+     * Convert object to BigDecimal.
+     *
+     * @param value current db object
+     * @param needScale need scale
+     * @param scale scale size
+     * @return big decimal
+     */
+    public static Object convertBigDecimalValue(final Object value, final boolean needScale, final int scale) {
+        if (null == value) {
+            return convertNullValue(BigDecimal.class);
+        }
+        if (value.getClass() == BigDecimal.class) {
+            return adjustBigDecimalResult((BigDecimal) value, needScale, scale);
+        }
+        if (value instanceof Number || value instanceof String) {
+            BigDecimal bigDecimal = new BigDecimal(value.toString());
+            return adjustBigDecimalResult(bigDecimal, needScale, scale);
+        }
+        throw new ShardingSphereException("Unsupported Date type: BigDecimal value %s", value);
+    }
+  
+    private static BigDecimal adjustBigDecimalResult(final BigDecimal value, final boolean needScale, final int scale) {
+        if (needScale) {
+            try {
+                return value.setScale(scale);
+            } catch (ArithmeticException ex) {
+                // try this if above fails
```

Comment:
no need to round

Reply:
this is reference mysql  driver source code